raiden/raiden_event_handler.py

```diff
@@ -73,13 +75,153 @@
 )
 
 
-def unlock(
+def handle_unlock(
     raiden: "RaidenService",
-    payment_channel: PaymentChannel,
-    end_state: NettingChannelEndState,
-    participant: Address,
-    partner: Address,
-) -> None:
+    chain_state: ChainState,
+    channel_unlock_event,
+    our_address,
+    unlock
+):
+    canonical_identifier = channel_unlock_event.canonical_identifier
+    token_network_identifier = channel_unlock_event.canonical_identifier.token_network_address
+    channel_identifier = channel_unlock_event.canonical_identifier.channel_identifier
+    participant = channel_unlock_event.participant
+
+    assert raiden.wal, "The Raiden Service must be initialize to handle events"
+
+    payment_channel: PaymentChannel = raiden.chain.payment_channel(
+        canonical_identifier=canonical_identifier
+    )
+
+    channel_state = get_channelstate_by_token_network_and_partner(
+        chain_state=chain_state,
+        token_network_id=TokenNetworkID(token_network_identifier),
+        creator_address=our_address,
+        partner_address=participant,
+    )
+
+    if not channel_state:
+        # channel was cleaned up already due to an unlock
+        raise RaidenUnrecoverableError(
+            f"Failed to find channel state with partner:"
+            f"{to_checksum_address(participant)}, token_network:pex(token_network_identifier)"
+        )
+
+    our_address = channel_state.our_state.address
```

Comment:
this is a parameter, but we are setting it again here.

can you please explain why this is the case?